pkg/expression/expression.go

```diff
@@ -971,7 +971,11 @@ func evaluateExprWithNullInNullRejectCheck(ctx BuildContext, schema *Schema, exp
 			if err != nil {
 				return nil, false, err
 			}
-			args[i], nullFromSets[i] = res, nullFromSet
+			// Next, `NewFunction` will be executed, which means this expression will be evaluated.
+			// To ensure correct execution, the executor will make some modifications,
+			// such as constant folding, which may alter the return type.
```

Comment:
do you know where is the exact place that constant folding execution will change the field type. Maybe improper in-place ft modification should be regulared as well

Reply:
It has happen here. 

![image](https://github.com/user-attachments/assets/dddb8bf5-d31a-4141-8b18-df943b81d00b)

However, the impact of this is too large, and for normal execution, directly executing and changing the expression will not have any effect. However, afterthe execution of null rejected , we still needs to use this expression, which leads to the problem.